pkg/client/unversioned/testclient/simple/simple_testclient.go

```diff
@@ -199,23 +203,14 @@ func body(t *testing.T, obj runtime.Object, raw *string) *string {
 		if err != nil {
 			t.Errorf("unexpected encoding error: %v", err)
 		}
-		// TODO: caesarxuchao: we should detect which group an object belongs to
-		// by using the version returned by Schem.ObjectVersionAndKind() once we
-		// split the schemes for internal objects.
-		// TODO: caesarxuchao: we should add a map from kind to group in Scheme.
 		var bs []byte
-		if api.Scheme.Recognizes(testapi.Default.GroupVersion().WithKind(fqKind.Kind)) {
-			bs, err = testapi.Default.Codec().Encode(obj)
-			if err != nil {
-				t.Errorf("unexpected encoding error: %v", err)
-			}
-		} else if api.Scheme.Recognizes(testapi.Extensions.GroupVersion().WithKind(fqKind.Kind)) {
-			bs, err = testapi.Extensions.Codec().Encode(obj)
-			if err != nil {
-				t.Errorf("unexpected encoding error: %v", err)
-			}
-		} else {
-			t.Errorf("unexpected kind: %v", fqKind.Kind)
+		g, found := testapi.Groups[fqKind.GroupVersion().Group]
```

Comment:
Do you mean instead of using `api.Scheme.ObjectKind()` to discover the group version, we should pass a testapi.TestGroup as an input argument of the `body()` function? That's feasible, but that will bloat all test code. We will need to pass a testapi.TestGroup every time we construct a response, e.g., https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/pods_test.go#L47

Do you think that's worthwhile?